Makefile

```diff
@@ -62,3 +62,59 @@ release:
 	cp ./pkg/linode-bs/deploy/releases/linode-blockstorage-csi-driver-$(IMAGE_VERSION).yaml ./$(RELEASE_DIR)
 	sed -e 's/appVersion: "latest"/appVersion: "$(IMAGE_VERSION)"/g' ./helm-chart/csi-driver/Chart.yaml
 	tar -czvf ./$(RELEASE_DIR)/helm-chart-$(IMAGE_VERSION).tgz -C ./helm-chart/csi-driver .
+
+
+## --------------------------------------
+## Testing
+## --------------------------------------
+
+##@ Testing:
+
+TEST_IMAGE_TAG ?= $(shell git rev-parse --abbrev-ref HEAD)
+TEST_IMAGE_NAME ?= "linode/linode-blockstorage-csi-driver"
+K8S_VERSION ?= "v1.29.1"
+CAPI_VERSION ?= "v1.6.3"
+HELM_VERSION ?= "v0.2.1"
+CAPL_VERSION ?= "v0.3.1"
+
+# Setting unique cluster name
+TEST_CLUSTER_NAME ?= csi-driver-cluster-$(shell git rev-parse --short HEAD)
+
+.PHONY: test-image-tags
+test-image-tags:
+	@echo $(TEST_IMAGE_TAG)
+
+.PHONY: remote-cluster-deploy
+remote-cluster-deploy:
+	# Create a CAPL test cluster without CSI driver and wait for it to be ready
+	clusterctl generate cluster $(TEST_CLUSTER_NAME) \
+		--kubernetes-version $(K8S_VERSION) \
+		--infrastructure linode-linode:$(CAPL_VERSION) \
+		--flavor kubeadm-vpcless | yq 'select(.metadata.name != "$(TEST_CLUSTER_NAME)-csi-driver-linode")' | kubectl apply -f -
+	kubectl wait --for=condition=ControlPlaneReady  cluster/$(TEST_CLUSTER_NAME) --timeout=600s
+	clusterctl get kubeconfig $(TEST_CLUSTER_NAME) > test-cluster-kubeconfig.yaml
+
+	# Install CSI driver and wait for it to be ready
+	cat e2e/setup/linode-secret.yaml | envsubst | KUBECONFIG=test-cluster-kubeconfig.yaml kubectl apply -f -
+	hack/generate-yaml.sh $(TEST_IMAGE_TAG) $(TEST_IMAGE_NAME) |KUBECONFIG=test-cluster-kubeconfig.yaml kubectl apply -f -
+	KUBECONFIG=test-cluster-kubeconfig.yaml kubectl rollout status -n kube-system daemonset/csi-linode-node --timeout=600s
+	KUBECONFIG=test-cluster-kubeconfig.yaml kubectl rollout status -n kube-system statefulset/csi-linode-controller --timeout=600s
+
+
+.PHONY: local-deploy
+local-deploy:
+	ctlptl apply -f e2e/setup/ctlptl-config.yaml
+	clusterctl init \
+		--wait-providers \
+		--core cluster-api:${CAPI_VERSION} \
+		--addon helm:${HELM_VERSION} \
+		--infrastructure linode-linode:${CAPL_VERSION}
+
+.PHONY: cleanup-cluster
+cleanup-cluster:
+	-kubectl delete cluster $(TEST_CLUSTER_NAME)
```

Comment:
We also need to delete linodevpc. Rest remaining are deleted as we are doing `kind delete` after this.

Reply:
Yup. I was originally using default then VPCs were lingering. So I thought why not just to do vpcless since we are just running tests on these clusters then destroying them